GCC 13.2: warning: ISO C++ prohibits anonymous structs [-Wpedantic]

Hello. 



I'm getting the following warning [here](https://github.com/jrouwe/JoltPhysics/blob/8bd2c9a1babd6b2d0c66a721c0d1d87d257a7383/Jolt/Core/Color.h#L66).

```cpp
Jolt/Core/Color.h:67:17: warning: ISO C++ prohibits anonymous structs [-Wpedantic]
   67 |                 {
      |                 ^

```

Apparently anonymous structs in C++ is an MSVC extension. 😅

Commit: 8bd2c9a1babd6b2d0c66a721c0d1d87d257a7383

(I'm not sure what can be done about this to preserve the old API)

--- a/Jolt/Core/Core.h
+++ b/Jolt/Core/Core.h
@@ -307,6 +307,7 @@
 	JPH_GCC_SUPPRESS_WARNING("-Wcomment")														\
 	JPH_GCC_SUPPRESS_WARNING("-Winvalid-offsetof")												\
 	JPH_GCC_SUPPRESS_WARNING("-Wclass-memaccess")												\
+	JPH_GCC_SUPPRESS_WARNING("-Wpedantic")														\
 																								\
 	JPH_MSVC_SUPPRESS_WARNING(4619) /* #pragma warning: there is no warning number 'XXXX' */	\
 	JPH_MSVC_SUPPRESS_WARNING(4514) /* 'X' : unreferenced inline function has been removed */	\

